src/ol/layer/Base.js

```diff
@@ -16,7 +16,8 @@ import {clamp} from '../math.js';
 
 /**
  * @typedef {import("../ObjectEventType").Types|'change:extent'|'change:maxResolution'|'change:maxZoom'|
- *    'change:minResolution'|'change:minZoom'|'change:opacity'|'change:visible'|'change:zIndex'} BaseLayerObjectEventTypes
+ *    'change:minResolution'|'change:minZoom'|'change:opacity'|'change:visible'|'change:zIndex'|
+ *    'change:source'} BaseLayerObjectEventTypes
```

Comment:
This is the wrong place unfortunately. `ol/layer/Base` does not have a source, because it is also the base class for `ol/layer/Group`. You'd have to define this type in `ol/layer/Layer` instead.

Reply:
Oh, in this case I don't think there is a type mismatch. If I use `if (x instanceof ol.layer.Layer)` in our code, it compiles fine.